openqaoa/backends/simulators/qaoa_analytical_sim.py

```diff
@@ -0,0 +1,111 @@
+#   Copyright 2022 Entropica Labs
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""
+Energy expectation as a function of angles computed accordingly to the analytical expression for p=1.
+"""
+import numpy as np
+
+from ...basebackend import QAOABaseBackend, QuantumCircuitBase
+from ...qaoa_parameters.baseparams import QAOAVariationalBaseParams
+from ...qaoa_parameters.operators import Hamiltonian
+from openqaoa.utilities import energy_expectation_analytical
+
+# not sure about those ones:
+from ...qaoa_parameters.baseparams import QAOACircuitParams, QAOAVariationalBaseParams 
+from typing import Union, List, Tuple, Type, Optional
+
+
+class QAOABackendAnalyticalSimulator(QAOABaseBackend):
+    r"""
+    A simulator class, specific for QAOA with a single layer, p=1, starting with a layer of Hadamards and using the X mixer. 
+    Works by calculating the expectation value of the given quantum circuit (specificied with beta and gamma angles) from the analytical formula derived in arXiv:2011.13420v2.
+     
+     Parameters
+    ----------
+    circuit_params: QAOACircuitParams
+        An object of the class ``QAOACircuitParams`` which contains information on 
+        circuit construction and depth of the circuit. Note that it only works for p=1 and the X Mixer Hamiltonian.
+    """
+    
+    def __init__(self, circuit_params: QAOACircuitParams, 
+                 prepend_state = None,
```

Comment:
These parameters are not used, I think `def __init__(self, circuit_params: QAOACircuitParam):` should be enough, otherwise, it can be confusing.